### ISSUE TITLE
moving to sparse matrix format

### ISSUE THREAD

**Issue by timydaley** [timydaley/scABC#7]
@linzx06 
I've run into difficulties applying scABC to data coming from the new 10X machines.  It's too big!  I'm trying to move scABC to use sparse matrices to save space and memory.  I've done some tweaks, specifically to reading in the data and computing the spearman correlation (see https://github.com/timydaley/SparseSpearmanCorrelation for my solution for computing the spearman correlation for sparse matrices).  Can you check the computation of the p-values?  I think the only place you use the matrix is in computing the crossprod, and the Matrix package has a built-in overloaded crossprod function.  I think the only thing we need to do is import that function.  Can you check this over when you get a chance?
Thanks.

**Comment by linzx06** [comment]
Hi Tim,

The matrix mu is the same dimension as the data matrix, and it is not
sparse. That one can cause some trouble too. Is it an issue of memory?

Best,
Z

**Comment by timydaley** [comment]
Yes it's an issue of memory. But mu should not be sparse, right?

**Comment by timydaley** [comment]
I'll test out all the parts and make sure everything works.  But this is good knowledge for what I need to look for.  Thanks Z.

**Comment by linzx06** [comment]
Sure Tim. Let me know if you need anything else.

Best,
Z

**Comment by MahdiZ11** [comment]
Hi Tim, I think we moved to sparse matrix format. Right?

**Comment by timydaley** [comment]
Not finished yet.  Had to work on other stuff.  I created a stable version before I messed around with the sparse matrix format.  Look under the release tab.

**Comment by MahdiZ11** [comment]
Okay. Thanks!
